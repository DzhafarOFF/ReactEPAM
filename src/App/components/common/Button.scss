.button{
    padding: 10px;
    margin: 10px;
    background-color: #E54B4B;
    color: #fefefe;
    border: 0;
    border-radius: 5%;
    text-transform: uppercase;
    font-weight: bolder;
    transition: 0.2s;
    cursor: pointer;
}
button:hover {
 background-color: lighten($color: #E54B4B, $amount: 5);
}
.button:disabled {
    background-color: grey;
}