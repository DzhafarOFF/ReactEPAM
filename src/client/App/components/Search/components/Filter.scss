.filter-title{
    display: inline-block;
}
.filter-option{
    display: inline-block;
}