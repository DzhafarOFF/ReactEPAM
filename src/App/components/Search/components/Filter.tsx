import React from 'react';
import Button from '../../common/Button'

type FilterProps = {
    options: string[];
    activeOption: string;
    handleSwitch: (e: React.MouseEvent) => void;
}
const Filter = (props: FilterProps) => {
  const setFilterOptions = props.options.map(
    (option, index) => {
      if (option === props.activeOption) return (<Button
        key = {index}
        content = {option}
        action = {props.handleSwitch}
        disabled = {true}
      />);
      return (<Button
        key = {index}
        content = {option}
        action = {props.handleSwitch}
      />);
    }
  );
  return (
      <>
        <h3>Sort by</h3>
        <div>  
          {setFilterOptions}
        </div>
      </>
  );
};

export default Filter;
