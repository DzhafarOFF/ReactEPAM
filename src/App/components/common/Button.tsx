import React from 'react';
import './Button.scss';

type ButtonsOptions = {
    content: string;
    action: (e: React.MouseEvent) => void;
    value?: string;
    disabled?: boolean;
}

const Button = (props: ButtonsOptions) => {
    return (
        <button
            className='button'
            onClick={props.action}
            value={props.content}
            disabled={props.disabled}>{props.content}
        </button>
    )
}

export default Button;