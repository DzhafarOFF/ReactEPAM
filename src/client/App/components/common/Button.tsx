import './Button.scss';
import React from 'react';

type Props = {
	content: string
	action: (e: React.MouseEvent) => void
	value?: string
	disabled?: boolean
};

const Button: React.FC<Props> = (props: Props) =>
	<button
		className='button'
		onClick={props.action}
		value={props.content}
		disabled={props.disabled}>{props.content}
	</button>
;


export default Button;
