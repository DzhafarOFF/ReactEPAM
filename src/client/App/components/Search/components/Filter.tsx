import './Filter.scss';
import Button from '../../common/Button';
import React from 'react';

type Props = {
	options: string[]
	activeOption: string
	handleSwitch: (e: React.MouseEvent) => void
};
const Filter: React.FC<Props> = (props: Props) => {
	const setFilterOptions = props.options.map(
		(option, index) => {
			if (option === props.activeOption) {
				return <Button
					key = {index}
					content = {option}
					action = {props.handleSwitch}
					disabled = {true}
				/>;
			}
			return <Button
				key = {index}
				content = {option}
				action = {props.handleSwitch}
				disabled = {false}
			/>;
		}
	);
	return (
		<div className = 'filter-option'>
			<h3 className='filter-title'>{'Search by'}</h3>
			{setFilterOptions}
		</div>
	);
};

export default Filter;
