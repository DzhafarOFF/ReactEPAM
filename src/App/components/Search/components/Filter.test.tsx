import Filter from './Filter';
import React from 'react';
import { shallow, render, mount } from 'enzyme';
import Button from '../../common/Button';

const handleSwitch = jest.fn()
const props = {
    options: ['title', 'genres'],
    activeOption: 'title',
    handleSwitch: handleSwitch
}

describe('<Filter/> rendering', () => {
    const filter = shallow(<Filter {...props} />);

    it('should renders correctly', () => {
        expect(filter).toMatchSnapshot();
    });

    it('should renders same numbers of buttons as options', () => {
        expect(filter.find(Button).length).toEqual(2);
    });

    it('should has one disabled button', () => {
        expect(filter.find(Button).findWhere(el => el.prop('disabled') === true).length).toEqual(1);
    }) 
})

describe('<Filter/> interaction', () => {
    const filter = mount(<Filter {...props} />);
    it('should handle switch event', () => {
        const btnActive = filter.find(Button).find('button').filterWhere(el => !el.prop('disabled'));
        btnActive.simulate('click');
        expect(handleSwitch.mock.calls.length).toEqual(1);
        // TODO:
        //expect(filter.find(Button).find('button').findWhere(el => el.prop('disabled') === false).length).toEqual(0);
    }) 
})