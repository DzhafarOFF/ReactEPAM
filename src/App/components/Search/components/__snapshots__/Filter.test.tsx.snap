// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Filter/> rendering should renders correctly 1`] = `
<div
  className="filter-option"
>
  <h3
    className="filter-title"
  >
    Search by
  </h3>
  <Button
    action={[MockFunction]}
    content="title"
    disabled={true}
    key="0"
  />
  <Button
    action={[MockFunction]}
    content="genres"
    key="1"
  />
</div>
`;
